File: java/hello/src/com/wingle/hello/thread/future/Hello.java
package com.wingle.hello.thread.future;

import com.wingle.hello.thread.util.SleepUtil;

import java.util.ArrayList;
import java.util.List;
import java.util.concurrent.*;

public class Hello {

    private static class Task implements Callable<String> {

        @Override
        public String call() throws Exception {
            SleepUtil.sleep(2);

            String tid = String.valueOf(Thread.currentThread().getId());
            System.out.println("Thread#" + tid + " : is calling...");

            return "threadid: " + tid;
        }
    }

    public static void main(String[] args) throws InterruptedException, ExecutionException {
        List<Future<String>> results = new ArrayList<Future<String>>();
        ExecutorService es = Executors.newCachedThreadPool();

        for(int i=0; i<100; i++) {
            Future<String> result = es.submit(new Task());
            results.add(result);
        }

        for(Future<String> result : results) {
            System.out.println("get result: " + result.get());
        }
    }
}

File: c/hello/hello.cpp
#include <iostream>

using namespace std;

int main() {
    cout << "Hello Cpp!" << endl;

    int i,j;
    cout << "未初始化的int i 的值是：" << i << endl;
    
    j = 1;
    j = i + j;
    cout << "is it correct" << j << endl;

    i = 10;
    cout << "初始化后的int i 的值是：" << i << endl;


    string s;
    cout << "未初始化的int s 的值是：" << s << endl;

    s = "wingle";
    cout << "初始化后的int s 的值是：" << s << endl;

    return 0;
}
